src/lookup.rs

```diff
@@ -147,14 +147,16 @@ impl App {
             let path = utils::uri_path(&file)?;
             node = match self.files.entry((**file).clone()) {
                 Entry::Occupied(entry) => {
-                    let (ast, _code) = entry.get();
+                    let (ast, _code, _) = entry.get();
                     ast.root().inner()?.clone()
                 }
                 Entry::Vacant(placeholder) => {
                     let content = fs::read_to_string(&path).ok()?;
                     let ast = rnix::parse(&content);
                     let node = ast.root().inner()?.clone();
-                    placeholder.insert((ast, content));
+                    let gc_root = Gc::new(Scope::Root(path));
+                    let evaluated = Expr::parse(node.clone(), gc_root);
```

Comment:
(nit) this should be called `parsed`, not `evaluated` here, correct? AFAICS the evaluation happens later :)

Reply:
Yep, good catch!